clippy_lints/src/matches.rs

```diff
@@ -322,6 +344,45 @@ fn check_overlapping_arms(cx: &LateContext, ex: &Expr, arms: &[Arm]) {
     }
 }
 
+fn check_wild_err_arm(cx: &LateContext, ex: &Expr, arms: &[Arm]) {
+    let ex_ty = walk_ptrs_ty(cx.tables.expr_ty(ex));
+    if match_type(cx, ex_ty, &paths::RESULT) {
+        for arm in arms {
+            if let PatKind::TupleStruct(ref path, ref inner, _) = arm.pats[0].node {
+                let path_str = print::to_string(print::NO_ANN, |s| s.print_qpath(path, false));
+                if inner.iter().any(|pat| pat.node == PatKind::Wild) &&
+                    path_str == "Err" {
+                        // `Err(_)` arm found
+                        let mut need_lint = true;
+                        if let ExprBlock(ref block) = arm.body.node {
+                            if is_unreachable_block(cx, block) {
+                                need_lint = false;
```

Comment:
Please use the `if_let_chain` macro to reduce rightward drift instead of mutable bools.